src/solvers/td3.ml

```diff
@@ -719,6 +719,44 @@ module WP =
             ) marked_for_deletion
         );
 
+        (* [destabilize_leaf] is meant for restarting of globals selected by the user. *)
+        (* Must be called on a leaf! *)
+        let destabilize_leaf (x : S.v) =
+          let destab_side_dep (x : S.v) =
+            match HM.find_option side_dep x with
+            | Some w ->
+              HM.remove side_dep x;
+              VS.iter (fun y ->
+                  HM.remove stable y;
+                  destabilize y;
+                ) w;
+            | None ->
+              ignore @@ Pretty.printf
+                "Could not find globals in side_dep, so side_dep %a is not restarted.\n"
+                S.Var.pretty_trace x
```

Comment:
It is restarted though: `restart_leaf` is called below regardless of this case. But I guess then the side effects are not retriggered and the analysis becomes unsound?

Reply:
The warning is indeed unnecessary, because there may be globals that are never written by a function, but have an initializer; so I removed the warning.